kuma/static/js/components/user-signup/auth-modal.js

```diff
@@ -3,38 +3,44 @@
 
     var authModalContainer = document.getElementById('auth-modal');
 
-    if (!window.mdn.modalDialog.showModal(authModalContainer)) {
+    if (
+        !authModalContainer ||
+        !window.mdn.multiAuthEnabled ||
+        !window.mdn.modalDialog.shouldShowModal()
+    ) {
         return;
     }
 
-    /* The React header takes just a couple of milliseconds to
-       long to load so, we need to wrap the code below in a 
-       setTimeout */
-    setTimeout(function() {
-        var pageHeader = document.querySelector('.page-header');
-        pageHeader.addEventListener('click', function(event) {
-            if (event.target.classList.contains('signin-link')) {
-                event.preventDefault();
-                var modalContentContainer = authModalContainer.querySelector(
-                    'section'
-                );
-                var closeModalButton = document.getElementById('close-modal');
+    var pageHeader = document.querySelector('.page-header');
+    pageHeader.addEventListener('click', function(event) {
+        function handleKeyup(event) {
+            if (event.key === 'Escape') {
+                closeModalButton.click();
+            }
+        }
+
+        var closeModalButton = document.getElementById('close-modal');
 
-                authModalContainer.classList.remove('hidden');
-                modalContentContainer.focus();
+        if (event.target.classList.contains('signin-link')) {
+            event.preventDefault();
+            var modalContentContainer = authModalContainer.querySelector(
+                'section'
+            );
 
-                closeModalButton.addEventListener('click', function() {
-                    window.mdn.modalDialog.closeModal(
-                        authModalContainer,
-                        event.target
-                    );
-                });
+            authModalContainer.classList.remove('hidden');
+            modalContentContainer.focus();
 
-                window.mdn.modalDialog.handleKeyboardEvents(
+            closeModalButton.addEventListener('click', function() {
+                window.mdn.modalDialog.closeModal(
                     authModalContainer,
                     event.target
                 );
-            }
-        });
-    }, 100);
+                document.removeEventListener('keyup', handleKeyup);
+            });
+
+            window.mdn.modalDialog.handleKeyboardEvents(authModalContainer);
+
+            document.addEventListener('keyup', handleKeyup);
```

Comment:
I am not sure about moving this here? It seems the functionality we would want with modals is to always listen on the `body` so, would it not be better to have `modal.js` take care of that?

Reply:
You technically can't. You need to make sure you execute the `document.body.removeEventListener('keyup', sameHandlerFunction)` when the modal closes. You don't get access to that inside `modal.js` because the closing is controlled from this file. 

It's not like `React.Component` which gets a `componentWillUnmount` so it itself would know when it's closing down.